resources/files/sacct/slurm_rest_api.rb

```diff
@@ -0,0 +1,148 @@
+require 'json'
+return if node['cluster']['node_type'] != 'HeadNode'
+
+slurm_etc = '/opt/slurm/etc'
+state_save_location = '/var/spool/slurm.state'
+key_location = state_save_location + '/jwt_hs256.key'
+certs_location = '/etc/ssl/certs'
+key_and_crt_name = 'nginx-selfsigned'
+id = 2005
+
+platform = node['platform']
+if platform == 'amazon'
+  platform = 'amzn2'
+end
+
+# Configure Slurm for JWT authentication and enable slurmrestd
+ruby_block 'Create JWT key file' do
+  block do
+    shell_out!("dd if=/dev/random of=#{key_location} bs=32 count=1")
+  end
+end
+
+# TODO: Not idempotent if user is in process 
+group 'slurmrestd' do
+    comment 'slurmrestd group'
+    gid id
+    system true
+end
+
+user 'slurmrestd' do
+  comment 'slurmrestd user'
+  uid id
+  gid id
+  home '/home/slurm'
+  system true
+  shell '/bin/bash'
+end
+
+file key_location do
+  owner 'slurm'
+  group 'slurm'
+  mode '0600'
+end
+
+directory state_save_location do
+  owner 'slurm'
+  group 'slurm'
+  mode '0755'
+end
```

Comment:
why we need to change permissions on this directory?

Reply:
Not entirely sure but this is consistent with the documentation [here](https://slurm.schedmd.com/jwt.html)